packages/headless/src/controllers/knowledge/generated-answer/headless-answerapi-generated-answer.test.ts

```diff
@@ -175,4 +189,32 @@ describe('knowledge-generated-answer', () => {
       expectedArgs
     );
   });
+
+  describe('subscribeToSearchRequest', () => {
+    it('triggers a fetchAnswer only when there is a request id, a query, and the request is made with another request than the last one', () => {
```

Comment:
I think this would be better as well

Reply:
I tend to agree with the test separation. And wouldn't do the test like this if it was easy to do it otherwise. 
Its not possible to reset the `vi.mock` before each test. That kind of mock cannot work with global state variables + cannot be declared in the test scope.
Any example in headless on how you would do this would be appreciated.

Reply:
top of my head, I can't think of a workaround

let's go with this for now, then!